internal/venice-common/src/main/java/com/linkedin/venice/authentication/AuthenticationService.java

```diff
@@ -0,0 +1,30 @@
+package com.linkedin.venice.authentication;
+
+import com.linkedin.venice.authorization.Principal;
+import com.linkedin.venice.utils.VeniceProperties;
+import java.io.Closeable;
+import java.security.cert.X509Certificate;
+
+
+/**
+ * Performs authentication.
+ */
+public interface AuthenticationService extends Closeable {
+  default void initialise(VeniceProperties veniceProperties) throws Exception {
+  }
+
+  @Override
+  default void close() {
+  }
+
+  default Principal getPrincipalFromHttpRequest(HttpRequestAccessor requestAccessor) {
+    return null;
+  }
+
+  interface HttpRequestAccessor {
+    String getHeader(String headerName);
+
+    X509Certificate getCertificate();
```

Comment:
I don't see any code calling this function... is it needed?

Reply:
This is here:
- to show that it can access the X509Certificate
- to show a path to implement an AuthenticationService based on X509Certificate (that is the current behaviour)

Reply:
let's keep it